src/builders/thermodynamic.jl

```diff
@@ -0,0 +1,252 @@
+"""
+$(TYPEDSIGNATURES)
+
+Build a max min driving force analysis model. See the docstring of
+[`max_min_driving_force_analysis`](@ref) for details about the arguments.
+"""
+function build_max_min_driving_force_model(
+    model::A.AbstractFBCModel,
+    reaction_standard_gibbs_free_energies::Dict{String,Float64};
+    reference_flux = Dict{String,Float64}(),
+    proton_ids::Vector{String},
+    water_ids::Vector{String},
+    concentration_lb = 1e-9, # M
+    concentration_ub = 1e-1, # M
+    T = 298.15, # Kelvin
+    R = 8.31446261815324e-3, # kJ/K/mol
+    ignore_reaction_ids = String[],
+)
+    # check if reactions that will be used in the model all have thermodynamic data, otherwise throw error.
+    # only these reactions will form part of the model.
+    rxn_source =
+        isempty(reference_flux) ? A.reactions(model) : collect(keys(reference_flux))
+    rxns = Set(x for x in rxn_source if !(x in ignore_reaction_ids))
+    isempty(setdiff(rxns, keys(reaction_standard_gibbs_free_energies))) || throw(
+        DomainError(
+            reaction_standard_gibbs_free_energies,
+            """
+                      Not all reactions have thermodynamic data.
+                      Either add the reactions with missing ΔG0s to ignore_reaction_ids,
+                      or add the data to reaction_standard_gibbs_free_energies.
+                      """,
+        ),
+    )
```

Comment:
I originally thought this checks whether all reactions from the model are covered, while in fact it only checks the ones from gibbs energy dics vs reference flux dict........ Better just check directly whether the keysets are equal.

...or maybe go with something that is correct by construction?

Reply:
So we don't actually care if all the model reactions have data (this will almost never be the case), neither do I think its necessary to always feed in the correct model (overkill). Rather, just make sure that the reactions that end up in the MMDF model have all the required data. The error specifies this